p2p/transport/webrtc/util.go

```diff
@@ -0,0 +1,134 @@
+package libp2pwebrtc
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"math/rand"
+	"strings"
+	"sync"
+
+	pb "github.com/libp2p/go-libp2p/p2p/transport/webrtc/pb"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/multiformats/go-multibase"
+	mh "github.com/multiformats/go-multihash"
+	"github.com/pion/datachannel"
+	"github.com/pion/webrtc/v3"
+	"google.golang.org/protobuf/proto"
+)
+
+func fingerprintToSDP(fp *mh.DecodedMultihash) string {
+	if fp == nil {
+		return ""
+	}
+	fpDigest := intersperse2(hex.EncodeToString(fp.Digest), ':', 2)
+	return getSupportedSDPString(fp.Code) + " " + fpDigest
```

Comment:
TODO: below can be a subtle bug as `getSupportedSDPString` can return
an empty string due to an error, but as we now append it no longer is an empty string,
thus probably lead do an obscure error much later?
Not better to just return an error to begin with?!

Reply:
fixed in https://github.com/libp2p/go-libp2p/pull/1999/commits/f76dfc499857bffe287bedd931eeba6c652a4825 (first commit of new PR https://github.com/libp2p/go-libp2p/pull/1999)